packages/@aws-cdk/aws-cognito/test/user-pool-client.test.ts

```diff
@@ -42,6 +42,57 @@ describe('User Pool Client', () => {
     expect(() => client2.userPoolClientName).toThrow(/available only if specified on the UserPoolClient during initialization/);
   });
 
+  describe('Client with secret', ()=>{
+
+    test('generate secret', ()=>{
+      // GIVEN
+      const stack = new Stack();
+      const pool = new UserPool(stack, 'Pool');
+
+      // WHEN
+      const clientWithSecret = new UserPoolClient(stack, 'clientWithSecret', {
+        userPool: pool,
+        generateSecret: true,
+      });
+
+      // THEN
+      expect(clientWithSecret.userPoolClientSecret).toBeDefined();
```

Comment:
We are doing it literally one line below.

Reply:
Not really. They test is testing that the output, whatever it is, is the same when the function is called multiple times. I wanted this test to test against the expected values we should see in the template. While you could argue that the integration tests are doing that, it's within a much larger context and there's a lot more room to miss that the output isn't what we expected.

Reply:
@TheRealAmazonKendra the final CloudFormation template looks like this:
```
      {
        "Type": "AWS::Cognito::UserPoolClient",
        "Properties": {
          "UserPoolId": {
            "Ref": "PoolD3F588B8"
          },
          "AllowedOAuthFlows": [
            "implicit",
            "code"
          ],
          "AllowedOAuthFlowsUserPoolClient": true,
          "AllowedOAuthScopes": [
            "profile",
            "phone",
            "email",
            "openid",
            "aws.cognito.signin.user.admin"
          ],
          "CallbackURLs": [
            "https://example.com"
          ],
          "GenerateSecret": true,
          "SupportedIdentityProviders": [
            "COGNITO"
          ]
        }
      }

```

It doesn't include the generated Client secret, since that one is generated by AWS *after* the Client is created by CloudFormation. So there is no way to retrieve its value in a unit test context.

I guess you were thinking of code resembling the one below, but we simply have nothing to put instead of the `xxx` in that code:
```ts
Template.fromStack(stack).hasResourceProperties('AWS::Cognito::UserPoolClient', {
        xxx: Match.exact(clientWithSecret.userPoolClientSecret),
      });
```

Reply:
That's fine. What I want to see tested against is what we put into the template on synth. This is to make sure that (a) the new behavior works as intended and (b) it is not accidentally undone by future changes.

Reply:
I see. You are referring to tests that check for the existence and correctness of the "GenerateSecret" field in the generated CloudFormaton template. I've now created tests to check for this. Back to you... :)